tensorflow/contrib/seastar/seastar_client_tag.cc

```diff
@@ -0,0 +1,273 @@
+#include "tensorflow/core/common_runtime/dma_helper.h"
+#if GOOGLE_CUDA
+#include "tensorflow/core/common_runtime/gpu/gpu_util.h"
+#include "tensorflow/core/common_runtime/gpu/process_state.h"
+#endif // GOOGLE_CUDA
+#include "tensorflow/core/distributed_runtime/worker_env.h"
+#include "tensorflow/core/lib/core/threadpool.h"
+#include "tensorflow/core/platform/logging.h"
+
+#include "tensorflow/contrib/seastar/seastar_client_tag.h"
+#include "tensorflow/contrib/seastar/seastar_message.h"
+#include "tensorflow/contrib/seastar/seastar_tensor_coding.h"
+
+namespace tensorflow {
+namespace {
+void ProcessCallOptions(SeastarClientTag* tag) {
+  if (tag->call_opts_ != nullptr) {
+    if (tag->call_opts_->GetTimeout() > 0) {
+      tag->timeout_in_ms_ = tag->call_opts_->GetTimeout();
+    }
+  }
+}
+} // namespace
+
+void InitSeastarClientTag(protobuf::Message* request,
+                          protobuf::Message* response,
+                          StatusCallback done,
+                          SeastarClientTag* tag,
+                          CallOptions* call_opts) {
+  tag->req_body_buf_.len_ = request->ByteSize();
+  tag->req_body_buf_.data_ = new char[tag->req_body_buf_.len_]();
+  request->SerializeToArray(tag->req_body_buf_.data_, tag->req_body_buf_.len_);
+
+  tag->req_header_buf_.len_ = SeastarClientTag::HEADER_SIZE;
+  tag->req_header_buf_.data_ = new char[SeastarClientTag::HEADER_SIZE];
+  
+  memcpy(tag->req_header_buf_.data_, "DEADBEEF", 8);
+  memcpy(tag->req_header_buf_.data_ + 8, &tag, 8);
+  memcpy(tag->req_header_buf_.data_ + 16, &tag->method_, 4);
+  // memcpy(tag->req_header_buf_.data_ + 20, &tag->status_, 2);
```

Comment:
Is this commented-out line needed? Can it be removed?

Reply:
Yes, removed, thx.